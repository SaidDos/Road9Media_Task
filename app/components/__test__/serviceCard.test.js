import 'react-native';
import React from 'react';
import ServiceCard from '../cards/serviceCard';

import renderer from 'react-test-renderer';

it('should renders correctly and generating a snapShot', () => {
  let service = {
    id: 1,
    name: 'someName',
    image: 'someImage',
    review: {overAllRating: 5},
  };
  const tree = renderer.create(<ServiceCard service={service} />).toJSON();
  expect(tree).toMatchSnapshot();
});
