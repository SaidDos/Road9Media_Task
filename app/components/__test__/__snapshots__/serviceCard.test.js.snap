// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly and generating a snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": 300,
      "marginBottom": 20,
      "width": 710,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "someImage",
      }
    }
    style={
      Object {
        "height": "60%",
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 18,
        "fontWeight": "bold",
        "padding": 5,
      }
    }
  >
    someName
  </Text>
  <Text
    style={
      Object {
        "color": "lightgrey",
        "padding": 5,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "justifyContent": "flex-start",
        }
      }
    />
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#07dddd",
          "fontWeight": "bold",
          "position": "absolute",
          "right": 10,
        }
      }
    >
       SAR 
      <Text
        style={
          Object {
            "color": "grey",
          }
        }
      >
        / Hour
      </Text>
    </Text>
  </View>
</View>
`;
